src/lib/structures/Command.ts

```diff
@@ -0,0 +1,231 @@
+
+	/**
+	 * Number of re-prompts of an argument
+	 * @since 0.0.1
+	 */
+	public promptLimit: number;
+
+	/**
+	 * Time allowed for re-prompts
+	 * @since 0.0.1
+	 */
+	public promptTime: number;
+
+	/**
+	 * Whether to have flag support for the command
+	 * @since 0.0.1
+	 */
+	public flagSupport: boolean;
+
+	/**
+	 * Allow use of quoted strings for arguments
+	 * @since 0.0.1
+	 */
+	public quotedStringSupport: boolean;
+
+	/**
+	 * Which type of channel the command can execute in
+	 * @since 0.0.1
+	 */
+	public runIn: ChannelType[];
+
+	/**
+	 * Whether the command has sub-commands or not
+	 * @since 0.0.1
+	 */
+	public subcommands: boolean;
+
+	// todo - public usage: CommandUsage;
+
+	/**
+	 * Whether the cooldown applies to author, channel or guild
+	 * @since 0.0.1
+	 */
+	public cooldownLevel: CooldownLevel;
+
+	/**
+	 * The time and limit for cooldown
+	 * @since 0.0.1
+	 */
+	public cooldowns: RateLimitManager;
+
+	/**
+	 * @since 0.0.1
+	 * @param store The command store
+	 * @param directory The base directory to the pieces folder
+	 * @param file The path from the pieces folder to the command file
+	 * @param options Optional Command settings
+	 */
+	public constructor(store: CommandStore, directory: string, files: readonly string[], options: CommandOptions = {}) {
+		super(store, directory, files, options);
+		options = options as Required<CommandOptions>;
+
+		this.name = this.name.toLowerCase();
+		this.requiredPermissions = new Permissions(options.requiredPermissions).freeze();
+		this.deletable = options.deletable as boolean;
+
+		this.description = options.description as string;
+		this.extendedHelp = options.extendedHelp as string;
+
+		this.fullCategory = files.slice(0, -1);
+		this.guarded = options.guarded as boolean;
+		this.hidden = options.hidden as boolean;
+		this.nsfw = options.nsfw as boolean;
+		this.permissionLevel = options.permissionLevel as number;
+		this.promptLimit = options.promptLimit as number;
+		this.promptTime = options.promptTime as number;
+		this.flagSupport = options.flagSupport as boolean;
+		this.quotedStringSupport = options.quotedStringSupport as boolean;
+		this.runIn = options.runIn as ChannelType[];
+		this.subcommands = options.subcommands as boolean;
+		// todo - this.usage = new CommandUsage(this.client, options.usage as string, options.usageDelim as string, this);
+		this.cooldownLevel = options.cooldownLevel as CooldownLevel;
+		if (!['author', 'channel', 'guild'].includes(this.cooldownLevel)) throw new Error('Invalid cooldownLevel');
+		this.cooldowns = new RateLimitManager(options.cooldown as number, options.bucket as number);
+	}
+
+	/**
+	 * The main category for the command
+	 * @since 0.0.1
+	 * @readonly
+	 */
+	public get category(): string {
+		return this.fullCategory[0] || 'General';
+	}
+
+	/**
+	 * The sub category for the command
+	 * @since 0.0.1
+	 * @readonly
+	 */
+	public get subCategory(): string {
+		return this.fullCategory[1] || 'General';
```

Comment:
No out-of-bounds array access
```suggestion
		return this.fullCategory.length > 1 ? this.fullCategory[1] : 'General';
```